apollo-ios-codegen/Sources/IR/IR+SelectionSet.swift

```diff
@@ -160,3 +160,17 @@ public class SelectionSet: Hashable, CustomDebugStringConvertible {
   }
 
 }
+
+extension LinkedList where T == ScopeCondition {
+  var containsDeferredFragment: Bool {
```

Comment:
This walks the linked list backwards because it would make sense that the tail of the scope descriptor being evaluated is the fragment that is deferred, which should provide an early exit.